LukeZvada/flashcards-server
# Basic User Registration & PR Template

#### Changes Made
1. Added file `auth.py` to `views` directory and wrote logic for basic user registration.
1. Added file `users.json` and `tokens.json` to `fixtures` directory.
1. Added PR Template
1. Modified file `reset.sh` to include `loaddata` command for users/tokens fixtures.
1. Modified file `urls.py` to include appropriate url routes. 
1. Modified file `.gitignore` to also ignore `.vscode` files 
​
#### Steps to Review
1. Checkout this branch locally.
```
git fetch --all
git checkout mp-registration
```
2. Initialize virtual environment and load seed data
```
pipenv shell
./reset.sh
```
_If you get a permissions error, run `chmod u+x reset.sh`_

3. Run the server
```
python manage.py runserver
```
3. Test registration in Postman.
> Open Postman and create a new POST request to `http://localhost:8000/register`
> Copy and paste the code below into the request body (raw JSON):
```
{
        "username": "exampleuser",
        "email": "example@user.com",
        "password": "test",
        "first_name": "Example",
        "last_name": "User"
}
```
Expected response:
{
        "token": "JuMblEdUPM3sS0FCh4R4cTeRsANDNuM83RS"
}
4. View code file.
> Confirm file modifications are present as indicated above.
> Confirm no unused code or extraneous comments exist.
> Confirm documentation is present where necessary and appropriate.
